src/web-extension/options/components/form.test.jsx

```diff
@@ -7,145 +7,151 @@ import TemplateInput from './template-input';
 import { defaults as fallbacks, helpers } from '../../../common/format';
 
 describe('form', () => {
-  function render(overrides) {
-    const store = { get: jest.fn(), set: jest.fn() };
+  function render(overrides, options = {}) {
+    const store = { get: jest.fn().mockResolvedValue({}), set: jest.fn().mockResolvedValue({}) };
     const defaults = { store };
 
     const props = { ...defaults, ...overrides };
-    const wrapper = shallow(<Form {...props} />);
-    const instance = wrapper.instance();
+    const wrapper = shallow(<Form {...props} />, options);
 
-    return { wrapper, instance, props };
+    return wrapper;
   }
 
   const inputs = wrapper => wrapper.find(TemplateInput);
   const input = (wrapper, name) => inputs(wrapper).find({ name });
   const value = (wrapper, name) => input(wrapper, name).prop('value');
 
-  const change = (wrapper, instance, name, val) => {
+  const change = (wrapper, name, val) => {
     const event = { target: { name, value: val } };
-    instance.handleChanged(event);
-    wrapper.update();
+    const field = wrapper.find(TemplateInput).find({ name });
+    field.simulate('change', event);
   };
 
   it('renders a template-input for the branch name format', () => {
-    const { wrapper, instance } = render({});
-    expect(input(wrapper, 'branch').props()).toEqual({
+    const wrapper = render({});
+    const field = input(wrapper, 'branch');
+
+    expect(field.props()).toEqual({
       label: 'Branch Name Format',
       id: 'branch-name-format',
       name: 'branch',
       value: '',
       fallback: fallbacks.branch,
       disabled: true,
-      onChange: instance.handleChanged,
+      onChange: expect.any(Function),
```

Comment:
You could also use `wrapper.instance().handleChanged` if you want to be explicit.

Reply:
Good point. My idea with updating the tests was to make them less implementation-dependent and focus more on the effects of user interactions. At the same time it probably doesn't hurt to be more specific here. I'll update the test 🙇

Reply:
Updated the tests. Thanks @lwassermann :metal